src/tasks/DailyTaskItemRunner.py

```
@@ -0,0 +1,1357 @@
+        if not matches:
+            return {
+                "verified": True,
+                "task_completed": True,
+                "post_action_result": "gift_task_card_vanished",
+            }
+        after = matches[0]
+        before_progress = str(getattr(before_card, "progress_text", "") or "")
+        after_progress = str(getattr(after, "progress_text", "") or "")
+        after_action = str(getattr(after, "action", "") or "")
+        progress_changed = bool(before_progress and after_progress and before_progress != after_progress)
+        completed_progress = after_progress.startswith("1/") or after_progress.endswith("/1") and after_progress[:1] == "1"
```

Ruff (0.15.12)

[warning] Parenthesize `a and b` expressions when chaining `and` and `or` together, to make the precedence clear

Parenthesize the `and` subexpression

(RUF021)